beacon_node/beacon_chain/src/data_column_verification.rs

```diff
@@ -215,6 +215,36 @@ impl<T: BeaconChainTypes, O: ObservationStrategy> GossipVerifiedDataColumn<T, O>
         )
     }
 
+    /// Create a `GossipVerifiedDataColumn` from `DataColumnSidecar` for block production ONLY.
+    /// When publishing a block constructed locally, the EL will have already verified the cell proofs.
+    /// When publishing a block constructed externally, there will be no columns here.
+    pub fn new_for_block_publishing(
+        column_sidecar: Arc<DataColumnSidecar<T::EthSpec>>,
+        chain: &BeaconChain<T>,
+    ) -> Result<Self, GossipDataColumnError> {
+        // Check if the data column is already in the DA checker cache. This happens when data columns
+        // are made available through the `engine_getBlobs` method.  If it exists in the cache, we know
+        // it has already passed the gossip checks, even though this particular instance hasn't been
+        // seen / published on the gossip network yet (passed the `verify_is_first_sidecar` check above).
+        // In this case, we should accept it for gossip propagation.
```

Comment:
I think we should also check for `is_first_sidecar` here, and we can avoid acquiring the cache write lock below (in the `observe` fn) if not needed - the cache read & write happens in burst around this time and it's time sensitive so it'd be great to avoid this.

Reply:
yeah the rename sounds good, IIRC it was originally named to match the spec, but I agree it's confusing without the full context:
> [IGNORE] The sidecar is the first sidecar for the tuple (block_header.slot, block_header.proposer_index, sidecar.index) with valid header signature, sidecar inclusion proof, and kzg proof.